Userland/Libraries/LibJS/JIT/Compiler.cpp

```diff
@@ -1084,42 +1115,56 @@ void Compiler::compile_mul(Bytecode::Op::Mul const& op)
-                                                                                         \
-        void Compiler::compile_##snake_case_name(Bytecode::Op::TitleCaseName const& op)  \
-        {                                                                                \
-            load_vm_register(ARG1, op.lhs());                                            \
-            load_accumulator(ARG2);                                                      \
-                                                                                         \
-            Assembler::Label end {};                                                     \
-                                                                                         \
-            branch_if_both_int32(ARG1, ARG2, [&] {                                       \
-                m_assembler.sign_extend_32_to_64_bits(ARG1);                             \
-                m_assembler.sign_extend_32_to_64_bits(ARG2);                             \
-                                                                                         \
-                /* accumulator = SHIFTED_BOOLEAN_TAG | (arg1 condition arg2) */          \
-                m_assembler.mov(                                                         \
-                    Assembler::Operand::Register(GPR0),                                  \
-                    Assembler::Operand::Imm(SHIFTED_BOOLEAN_TAG));                       \
-                m_assembler.cmp(                                                         \
-                    Assembler::Operand::Register(ARG1),                                  \
-                    Assembler::Operand::Register(ARG2));                                 \
-                m_assembler.set_if(                                                      \
-                    Assembler::Condition::AssemblerCondition,                            \
-                    Assembler::Operand::Register(GPR0)); /* sets only first byte */      \
-                store_accumulator(GPR0);                                                 \
-                                                                                         \
-                m_assembler.jump(end);                                                   \
-            });                                                                          \
-                                                                                         \
-            native_call((void*)cxx_##snake_case_name);                                   \
-            store_accumulator(RET);                                                      \
-            check_exception();                                                           \
-            end.link(m_assembler);                                                       \
+#    define DO_COMPILE_COMPARISON_OP(TitleCaseName, snake_case_name, IntegerCondition, FloatCondition) \
```

Comment:
clang-format seems to get confused on this block and the formatting isn't great. If you do a bit of manual formatting first, clang-format does a much better job:
```c++
#    define DO_COMPILE_COMPARISON_OP(TitleCaseName, snake_case_name, IntegerCondition, FloatCondition) \
        static Value cxx_##snake_case_name(VM& vm, Value lhs, Value rhs)                               \
        {                                                                                              \
            return TRY_OR_SET_EXCEPTION(snake_case_name(vm, lhs, rhs));                                \
        }                                                                                              \
                                                                                                       \
        void Compiler::compile_##snake_case_name(Bytecode::Op::TitleCaseName const& op)                \
        {                                                                                              \
            load_vm_register(ARG1, op.lhs());                                                          \
            load_accumulator(ARG2);                                                                    \
                                                                                                       \
            compiler_comparison_fastpaths(                                                             \
                ARG1, ARG2,                                                                            \
                [&](auto lhs, auto rhs) {                                                              \
                    m_assembler.sign_extend_32_to_64_bits(lhs);                                        \
                    m_assembler.sign_extend_32_to_64_bits(rhs);                                        \
                                                                                                       \
                    /* accumulator = SHIFTED_BOOLEAN_TAG | (arg1 condition arg2) */                    \
                    m_assembler.mov(                                                                   \
                        Assembler::Operand::Register(GPR0),                                            \
                        Assembler::Operand::Imm(SHIFTED_BOOLEAN_TAG));                                 \
                    m_assembler.cmp(                                                                   \
                        Assembler::Operand::Register(lhs),                                             \
                        Assembler::Operand::Register(rhs));                                            \
                    m_assembler.set_if(                                                                \
                        Assembler::Condition::IntegerCondition,                                        \
                        Assembler::Operand::Register(GPR0)); /* sets only first byte */                \
                    return GPR0;                                                                       \
                },                                                                                     \
                [&](auto lhs, auto rhs) {                                                              \
                    /* accumulator = SHIFTED_BOOLEAN_TAG | (arg1 condition arg2) */                    \
                    m_assembler.mov(                                                                   \
                        Assembler::Operand::Register(GPR0),                                            \
                        Assembler::Operand::Imm(SHIFTED_BOOLEAN_TAG));                                 \
                    m_assembler.cmp(                                                                   \
                        Assembler::Operand::FloatRegister(lhs),                                        \
                        Assembler::Operand::FloatRegister(rhs));                                       \
                    m_assembler.set_if(                                                                \
                        Assembler::Condition::FloatCondition,                                          \
                        Assembler::Operand::Register(GPR0)); /* sets only first byte */                \
                    return lhs;                                                                        \
                },                                                                                     \
                [&](auto lhs, auto rhs) {                                                              \
                    m_assembler.mov(                                                                   \
                        Assembler::Operand::Register(ARG1),                                            \
                        Assembler::Operand::Register(lhs));                                            \
                    m_assembler.mov(                                                                   \
                        Assembler::Operand::Register(ARG2),                                            \
                        Assembler::Operand::Register(rhs));                                            \
                    native_call((void*)cxx_##snake_case_name);                                         \
                    return RET;                                                                        \
                });                                                                                    \
        }
```